src/packagedcode/rubygems.py

```diff
@@ -647,93 +648,124 @@ def normalize(gem_data, known_fields=known_fields):
     )
 
 
-def parse_spec(location):
-    pass
+# defaults fields of .gemspec file
+data_dic = {
+    "name": None,
+    "version": None,
+    "authors": None,
+    "email": None,
+    "summary": None,
+    "description": None,
+    "homepage": None,
+    "license": None,
+    "dependencies": None
+    }
 
 
-class GemSpec(object):
+def parse_spec(location):
+    """
+    Returns a dictionary which containing .gemspec file data.
+    """
+    file_data = []
+    with io.open(location, "r", encoding="utf-8") as f:
+        file_data = [line.rstrip('\n') for line in f]
+
+    dependencies = []
+
+    for individual in file_data:
+        # update the value of name
+        if 'name' in individual and 'basename' not in individual:
+            # >>> s = 'spec.name = "abc"'
+            # >>> name = s.split('= ')
+            # >>> name
+            # ['spec.name ', '"abc"']
+            # >>> name = name[1].strip('\'"')
+            # >>> name
+            # abc
+            name = individual.split('= ')
+            name = name[1].strip('\'"')
+            data_dic.update({'name' : name})
```

Comment:
Do we really need an update call here? why not just `data_dic['name'] = name` 

Also, `data_dic` is not a great variable name,

Reply:
@majurg You are right. We should use `data_dic['name] = name to update the value. This method is more faster than the previous one. 
References: https://stackoverflow.com/a/15456218/9380990